core/modules/views/plugins/views_plugin_display_block.inc

```diff
@@ -70,20 +70,29 @@ class views_plugin_display_block extends views_plugin_display {
    */
   function execute_hook_block_list($delta = 0, $edit = array()) {
     $delta = $this->view->name . '-' . $this->display->id;
-    $info = $this->get_option('block_description');
-
-    if (empty($info)) {
+    $block_admin_title = $this->get_option('block_description');
+	  $block_description = $this->get_option('block_layout_description');
+    // Check whether block admin title has been entered and
+    // if not, check whether the block display has been named.
+    if (empty($block_admin_title)) {
       if ($this->display->display_title == $this->definition['title']) {
-        $info = t('View: @view', array('@view' => t($this->view->get_human_name())));
+        // Block display has not been named so compose a title for display.
+        $block_admin_title = t('View: @view', array('@view' => t($this->view->get_human_name())));
       }
       else {
-        $info = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
+        // Block display has a name 'display title' so include this in display.
+        $block_admin_title = t('View: @view: @display', array('@view' => t($this->view->get_human_name()), '@display' => t($this->display->display_title)));
       }
     }
+    else {
+      // There is a block admin title so use this, prefixed with 'View'
+      // to indicate that this block originates from a view.
+      $block_admin_title = t('View: ') . $block_admin_title;
```

Comment:
I'm not sure about this either, and I prefer your tooltip suggestion. But I do find it helpful when looking in layouts at a lengthy list of blocks that many are prefixed 'field:' and some 'view:'.